src/main/java/ch/njol/skript/expressions/ExprFormatTime.java

```diff
@@ -51,35 +58,44 @@
 		Skript.registerExpression(ExprFormatTime.class, String.class, ExpressionType.PROPERTY, "%dates% formatted [human-readable] [(with|as) %-string%]");
 	}
 
-	@SuppressWarnings("null")
+	@SuppressWarnings("NotNullFieldNotInitialized")
 	private SimpleDateFormat format;
 
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<? extends String> customFormat;
+
 	@Override
-	@SuppressWarnings("null")
+	@SuppressWarnings({"null", "unchecked"})
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		setExpr((Expression<? extends Date>) exprs[0]);
-		if (exprs[1] != null) {
-			if (!(exprs[1] instanceof Literal)) {
-				VariableString str = (VariableString) exprs[1];
-				if (!str.isSimple()) {
-					Skript.error("Date format must not contain variables!");
-					return false;
-				}
+		customFormat = (Expression<? extends String>) exprs[1];
+
+		if (customFormat instanceof Literal || (customFormat instanceof VariableString && ((VariableString) customFormat).isSimple())) {
+			try {
+				format = new SimpleDateFormat(customFormat.getSingle(null));
+			} catch (Exception e) {
+				Skript.error("Incorrect date format used: " + e.getMessage());
```

Comment:
I think it'd be better worded as sth like `Invalid date format: <format>`. I'm also not sure if it'd be a good idea to provide the exception message, since it may be too technical, IMO it'd be better to just say it's invalid